java/lance-jni/src/blocking_dataset.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use arrow::array::RecordBatchReader;
+use tokio::runtime::Runtime;
+
+use lance::dataset::{Dataset, WriteParams};
+use lance::Result;
+
+pub struct BlockingDataset {
+    inner: Dataset,
+    rt: Runtime,
+}
+
+impl BlockingDataset {
+    pub fn write(
+        reader: impl RecordBatchReader + Send + 'static,
+        uri: &str,
+        params: Option<WriteParams>,
+    ) -> Result<Self> {
+        let rt = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()?;
```

Comment:
Could you move this to a lazy static variable? We don't want to create a new runtime every time we write to a dataset.

Note, however, in the future we should probably port the executor we did in Python #1172. This allows things like writing to a Lance dataset from a scan of another Lance dataset, and support for KeyboardInterrupt (#1438).